test/Constraints/members.swift

```diff
@@ -463,3 +463,6 @@ struct Outer {
     }
   }
 }
+
+// rdar://problem/39514009 - don't crash when trying to diagnose members with special names
+print("hello")[0] // expected-error {{}}
```

Comment:
You should make sure the error message is reasonable. More importantly now that 'init' is a special name we should make sure improper access of members named 'init' is diagnosed properly too.